docs/reactivity/overview.md

```diff
@@ -0,0 +1,240 @@
+
+Here you can find API with list of all available tools for reactivity and links to specific places in `Vue 3` docs: 👉 [Reactivity API](./api).
+
+Now, let's jump to some examples!
+
+## Building dialog system
+
+Let's say you have a lot of dialogs and would like to keep and manage them in one, global place. We can start with singleton class:
+
+```js
+let instance;
+
+export class DialogSystem {
+    constructor() {
+        this.root = document.createElement('div');
+        this.root.classList.add('dialog-root')
+
+        document.body.appendChild(this.root)
+    }
+
+    static getInstance() {
+        if (!instance) {
+            instance = new DialogSystem();
+        }
+
+        return instance;
+    }
+}
+```
+
+So, what's going on here? We have a singleton pattern, with static method `getInstance`. Every time, we try to call `DialogSystem.getInstance()`, we would check if `DialogSystem` instance exists and if not, we create one. Then return new instance or a cached one. With this approach, we ensure to always have one single instance.
+
+Then, in constructor we create new `div` element, and append it to `body`. We don't care about either styling or HTML structure, we are focusing on JS functionality.
+
+Next, we need to add a way to store new dialogs. Let's change out `constructor`!
+
+```js
+import { ref } from 'ovee.js';
+
+let instance;
+
+export class DialogSystem {
+    constructor() {
+        const root = document.createElement('div');
+
+        root.classList.add('dialog-root')
+        document.body.appendChild(document)
+
+        // store dialogs
+        this.dialogs = ref([])
+    }
+
+    /* ... */
+}
+```
+
+We used our first reactivity tool: `ref`. It creates reactive object with field `.value`. If reference under `.value` changes, we would now. Example:
+
+```js
+const arr = ref([1, 2, 3])
+
+arr.value = [1, 2, 3] // <- this will trigger a change!
+```
+
+Coming back to our example! Let's add a method, that allow us to create dialogs!.
+
+```js
+import { ref, markRaw } from 'ovee.js';
+
+export class DialogSystem {
+    /* ... */
+
+    // some options for dialog
+    createDialog(options, open = false) {
+        const dialog = new Dialog(options);
+
+        this.dialogs.push(markRaw(dialog)); // <- markRaw?!
+
+        if (open) {
+            dialog.open();
+        }
+
+        return dialog;
+    }
+
+    /* ... */
+}
+```
+
+So creating dialogs is quite simple: we create `Dialog` class instance (it's implemantation is irrelevant currently for us) and push it to the array. But what's about this `markRaw` method?!
+`ref` and `makeReactive` does a little bit of magic, that uwraps any nested `ref` inside it. To prevent it, we use `markRaw` to say: "We want to keep it that way, don't touch it!".
```

Comment:
```suggestion
`ref` and `makeReactive` do a little bit of magic, that unwraps any nested `ref` inside it. To prevent it, we use `markRaw` to say: "We want to keep it that way, don't touch it!".
```